### PR TITLE
Move v1/refs and v1/resource

### PR DESCRIPTION
**Pull request by caesarxuchao**
This PR moves pkg/api/v1/ref.go and pkg/api/v1/resource_helper.go to their own sub packages, it's very similar to 44299 and 44302.

The PR is mostly mechanical, except that 
* i moved some utility function from resource.go to pkg/api/v1/pod and pkg/api/v1/node, as they are more appropriate
* i updated the staging/copy.sh to copy the new subpackages, so that helper functions are copied. We can get rid of this copy after client-go stops copying API types.

### CODE REVIEW

File: pkg/api/v1/objectreference.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// IsAnAPIObject allows clients to preemptively get a reference to an API object and pass it to places that
+// intend only to get a reference to that object. This simplifies the event recording interface.
+func (obj *ObjectReference) SetGroupVersionKind(gvk schema.GroupVersionKind) {
+	obj.APIVersion, obj.Kind = gvk.ToAPIVersionAndKind()
+}
```
Comment (lavalamp) [inline pkg/api/v1/objectreference.go]: nit: missing blank line

File: pkg/api/v1/resource/resource_helpers.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
+/*
```
Comment (lavalamp) [inline pkg/api/v1/resource/resource_helpers.go]: nit: filename could just be "helpers" considering the package.

**Review by lavalamp — COMMENTED** [review]
Two small nits that aren't worth fixing unless you have to do a rebase.

**Comment by lavalamp** [comment]
It is kind of a disaster that we have multiple versions of all these functions, esp. considering that I want binaries to have their own copies of the internal types.

/lgtm
/approve

**Comment by caesarxuchao** [comment]
Addressed the comments and squashed. Thanks for the review.

**Comment by caesarxuchao** [comment]
> It is kind of a disaster that we have multiple versions of all these functions

Some are not needed for the internal versions and i had removed them in previous PRs.

**Pull request merged**